hal/src/electron/modem/mdm_hal.cpp

```diff
@@ -1110,7 +1174,7 @@ bool MDMParser::powerOff(void)
     LOCK();
     bool ok = false;
     bool continue_cancel = false;
-    bool check_ri = false;
+    bool check_ri = true;
```

Comment:
Why was this changed to `true` by default?  It's never set to `false`, only set true if the modem is R410.  IIRC we can only rely on RI_UC input as V_INT for R410, due to its hardware configuration.  This pin might not behave as expected for U2/G3.

Reply:
During testing we saw that we have to wait for the modem to actually turn off, otherwise it's very easy to catch it while it still performs the shutdown and it will respond with an OK to whatever command.

Reply:
Ok, based on our conversation this morning, you will be using the RX pin to check power state on U2/G3 and in that case we can remove this `check_ri` flag and just always check it.

Reply:
I'm going to make that change as part of https://github.com/particle-iot/device-os/pull/2197, since it's primarily focused on the power state detection/management and RX idea original came from there.

I'm also thinking that perhaps it's also fine if we rely on RI_UC for U2 as well, because there is no reliable way to check the modem type without first talking to it. On a related note, I'm going to at least remove the code that resets the device type to UNKNOWN every time we power on.

Reply:
And the only time RI_UC will become a problem is when `AT+URING` is used or we receive a call or SMS. Which isn't really the normal behavior.

Reply:
RI_UC (as RING, not V_INT) also toggles at a slow rate, so we can check it a few times to be sure it's in the right state if need be.

Reply:
:+1: Let me look into that.

Reply:
So, supposedly we can check two times with 1s in-between them to make sure we are not seeing RI being active as opposed to V_INT being down.

Reply:
Sounds good 👍